near-sdk/src/lib.rs

```diff
@@ -277,6 +277,63 @@ extern crate quickcheck;
 /// }
 /// ```
 ///
+/// ## `#[callback_unwrap]` (annotates function parameters)
+///
+/// Automatically unwraps the successful result of a callback from a cross-contract call.
+/// Used on parameters in callback methods that are invoked as part of a cross-contract call chain.
+/// If the promise fails, the method will panic with the error message.
+///
+/// This attribute is commonly used with [`PromiseOrValue<T>`], which represents either an immediate value
+/// or a promise that will resolve to that value.
+///
+/// ### Example with Cross-Contract Factorial:
+///
+/// ```rust
+/// # use near_sdk::{near, env, PromiseOrValue};
+/// #[near(contract_state)]
+/// #[derive(Default)]
+/// pub struct CrossContract {}
+///
+/// #[near]
+/// impl CrossContract {
+///     pub fn factorial(&self, n: u32) -> PromiseOrValue<u32> {
+///         if n <= 1 {
+///             return PromiseOrValue::Value(1);
+///         }
+///         let account_id = env::current_account_id();
+///         Self::ext(account_id.clone())
+///             .factorial(n - 1)
+///             .then(Self::ext(account_id).factorial_mult(n))
+///             .into()
+///     }
+///
+///     #[private]
```

Comment:
it can be mentioned with link to `#[private]` attribute's anchor that this is marked as private method to allow only calling 
`factorial_mult` from `factorial` contract method by `CrossContract` itself and disallow it to be called externally by users